File: client/src/common/models/group.ts
import { IActivity } from "./activity";
import { IPost } from "./post";

export interface IGroupsEnvelope {
  groups: IGroup[];
  groupCount: number;
}

export interface IGroup {
  id: string;
  title: string;
  description: string;
  category: string;
  dateCreated: Date;
  coverPhoto?: string;
  members: IGroupMember[];
  posts: IPost[];
  events: IActivity[];
}

export interface IGroupMember {
  username: string;
  displayName: string;
  image: string;
  isOwner: boolean;
}

File: Domain/Group.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Group
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Category { get; set; }
        public DateTime DateCreated { get; set; }
        public Photo CoverPhoto { get; set; }
        public ICollection<GroupMember> GroupMembers { get; set; }
        public ICollection<Post> Posts { get; set; }
        public ICollection<Activity> Activities { get; set; }
    }
}